web/packages/teleport/src/Roles/RoleList/RoleList.tsx

```diff
@@ -20,60 +20,76 @@ import React from 'react';
 import Table, { Cell } from 'design/DataTable';
 import { MenuButton, MenuItem } from 'shared/components/MenuAction';
 
-import { State as ResourceState } from 'teleport/components/useResources';
+import { SearchPanel } from 'shared/components/Search';
 
-import { State as RolesState } from '../useRoles';
+import { SeversidePagination } from 'teleport/components/hooks/useServersidePagination';
+import { RoleResource } from 'teleport/services/resources';
 
-export default function RoleList({
-  items = [],
-  pageSize = 20,
+export function RoleList({
   onEdit,
   onDelete,
-}: Props) {
+  onSearchChange,
+  search,
+  serversidePagination,
+}: {
+  onEdit(id: string): void;
+  onDelete(id: string): void;
+  onSearchChange(search: string): void;
+  search: string;
+  serversidePagination: SeversidePagination<RoleResource>;
+}) {
   return (
     <Table
-      data={items}
+      data={serversidePagination.fetchedData.agents}
+      fetching={{
+        fetchStatus:
+          serversidePagination.attempt.status === 'processing' ? 'loading' : '',
```

Comment:
this is kinda broken. i honestly don't remember how the ancient server side table used to work, there was also recent work on updating the table styles so it could be a regression.

on a slow network:
1) no loading state for the next/prev buttons, so i can spam click on them and make requests per click
2) similarly, when i use the search bar, my paging buttons still show, and i can click on it which also makes out requests

Reply:
you know what, the `newlock > roles list` loading state actually works correctly

Reply:
Good catch, thanks!
Actually `serversidePagination` has a `fetchStatus` field, that I should pass directly. This seems to fix one part of the issue.
As of the buttons not being disabled, that never worked 😬. I added a fix for this. 
I'm only not sure what do you mean here: 
>similarly, when i use the search bar, my paging buttons still show, and i can click on it which also makes out requests

but I believe it is fixed as well now?